src/dfm/cleaning/clean_cli.py

```diff
@@ -0,0 +1,324 @@
+from omegaconf import DictConfig, OmegaConf
+from tqdm import tqdm
+from tqdm.contrib.logging import logging_redirect_tqdm
+
+from dfm.cleaning import QualityFilter, SentenceFilter
+
+CFG_PATH = Path(__file__).parent / "configs"
+VALID_SAVE_FORMATS = {
+    "parquet": "parquet",
+    "json": "json",
+    "jsonl": "json",
+    "ndjson": "json",
+    "csv": "csv",
+    "arrow": "arrow",
+}
+
+
+def create_quality_filter(cfg):
+    cfg = cfg.quality_filter
+
+    qf = QualityFilter(
+        min_stop_words=cfg.min_stop_words,
+        mean_word_length=cfg.mean_word_length,
+        doc_length=cfg.doc_length,
+        alpha_ratio=cfg.alpha_ratio,
+        symbol_2_word_hashtag=cfg.symbol_2_word_hashtag,
+        symbol_2_word_ellipsis=cfg.symbol_2_word_ellipsis,
+        max_p_begin_bullets=cfg.max_p_begin_bullets,
+        max_p_end_ellipsis=cfg.max_p_end_ellipsis,
+        min_bullets=cfg.min_bullets,
+        min_ellipsis=cfg.min_ellipsis,
+        duplicate_lines_chr_fraction=cfg.duplicate_lines_chr_fraction,
+        duplicate_paragraph_chr_fraction=cfg.duplicate_paragraph_chr_fraction,
+        top_ngram_chr_fraction_thresholds=cfg.top_ngram_chr_fraction_thresholds,
+        top_ngram_chr_fraction_range=cfg.top_ngram_chr_fraction_range,
+        top_ngram_min_count=cfg.top_ngram_min_count,
+        duplicate_n_gram_fraction_thresholds=cfg.duplicate_n_gram_fraction_thresholds,
+        duplicate_n_gram_fraction_range=cfg.duplicate_n_gram_fraction_range,
+        max_length=cfg.max_length,
+        string_filter=cfg.string_filter,
+        language_detection_tool=cfg.language_detection_tool,
+        language_threshold=cfg.language_threshold,
+        languages=cfg.languages,
+        short_long_sentence_length_split=cfg.short_long_sentence_length_split,
+        short_long_sentence_threshold=cfg.short_long_sentence_threshold,
+        ignore_filters=cfg.ignore_filters,
+    )
+    return qf
+
+
+def create_sentence_filter(cfg):
+    cfg = cfg.sentence_filter
+    sf = SentenceFilter(
+        filter_names=cfg.filter_names,
+        title_cased_words_threshold=cfg.title_cased_words_threshold,
+        min_num_words=cfg.min_num_words,
+        curly_brackets_threshold=cfg.curly_brackets_threshold,
+        n_jobs=1,
+    )
+    return sf
+
+
+def q_filter(batch, cfg):
+    """
+    Quality filter which takes in a hf datasets batch and and applies a quality
+    filter to all the texts which pass the language filter
+    """
+    qf = create_quality_filter(cfg)
+
+    if cfg.save_meta_data:
+
+        valid_langs = set(cfg.valid_languages)
+        if valid_langs:
+
+            def filter_lang(batch, i):
+                return (
+                    batch[cfg.lang_col][i] in valid_langs
+                    and batch["passed_sentence_filter"]
+                )
+
+        is_filtered = [filter_lang(batch, i) for i, _ in enumerate(batch[cfg.text_col])]
+        texts = (t for t, is_f in zip(batch[cfg.text_col], is_filtered) if is_f)
```

Comment:
Missing a short comment describing what this is doing.

Reply:
Hopefully it is better now, otherwise let me know